src/shared/blockdev-util.h

```diff
@@ -14,6 +14,16 @@
 #define xsprintf_sys_block_path(buf, suffix, devno)                     \
         xsprintf(buf, "/sys/dev/block/%u:%u%s", major(devno), minor(devno), strempty(suffix))
 
+typedef enum BlockDeviceFlag {
+        BLOCK_DEVICE_WHOLE_DISK  = 1 << 0, /* whole block device, e.g. sda, nvme0n1, or loop0. */
+        BLOCK_DEVICE_BACKING     = 1 << 1, /* fd can be a file in a block device. */
```

Comment:
maybe change comment to something like: `fd may be regular file or directory on file system, in which case backing block device is determined`